src/orquestra/quantum/circuits/_circuit.py

```diff
@@ -165,6 +165,17 @@ def inverse(self) -> "Circuit":
                 " since there are operators in it without the `dagger` method."
             ) from e
 
+    def controlled(self):
+        """Return a circuit where all the operations from self are turned into their
+        respective controlled operations whose control is the 0th qubit."""
+        c_ops = []
+        for op in self.operations:
+            controlled_op = op.gate.controlled(1)
```

Comment:
It seems to me that the "controlled" index is always equal to "1" ?
I think it would be better if we had this index specified by user, or if `None` taking the "n_qubits+1".

Reply:
Sure I can add that feature, if you think it's worth it. I guess I just assumed it would be easy enough to just have the control be the first.

Reply:
But it's equal to "1" and not "0" – this is what threw me off :D

Reply:
Also – please add typing for `control_index` and we're good to go!

Reply:
Ugh! Thanks I keep forgetting the typing on this one. Thanks!